### ISSUE TITLE
XBox-type archive compatibility

### ISSUE THREAD

**Issue by Ethatron** [Ethatron/bsaopt#18]
Except XMem-codec for now. I guess it's raw LZX but I have to look a bit deeper for that.

**Comment by selyb** [comment]
The files/folders are being put in the wrong order. I think I misled you when I presented you with the new information and I am very sorry for that. It looks like the hashes are being created correctly but the hash needs to be byte swapped to determine the order.

First is archive.exe, second is bsaopt 2.0 beta
![Test1_archive](https://f.cloud.github.com/assets/1646249/15443/fed3c82e-476a-11e2-87cc-1305ee595d74.png)
![Test1_bsaopt2](https://f.cloud.github.com/assets/1646249/15445/068aac18-476b-11e2-9890-6912a0b4919f.png)

**Comment by Ethatron** [comment]
In reality I don't think the order is a matter of life-and-death, at least Oblivion-BSA I verified work with any order.
I added the swap, but let's see if there are other things coming up with the beta before I upload a new compile.

Can you verify the files work on an XBox? I don't have one, so I can't check.

**Comment by selyb** [comment]
Gimme 5 mins

**Comment by selyb** [comment]
Looks like it works. I have merged Update.bsa, Update2.bsa, Dawnguard,bsa, and Hearthfires.bsa and playing on a new game looks completely functional

**Comment by Ethatron** [comment]
Cool. I saw your post on XBox-Chaos, seems it just got easier. :)

**Comment by selyb** [comment]
Yeah, I wanted to confirm it working before I updated my tutorials. This functionality will make it much much easier for many n00bs :)

**Comment by Ethatron** [comment]
Do have Oblivion on XBox as well? I haven't any information on that, and there is no "Archive.exe" to check XBox BSA's with.
And DDX-files? I could add that to DDSopt as part of the XBox-compatibility scheme.

**Comment by selyb** [comment]
I do have Oblivion, I just never got into the game so I never got into modding it either. I can put it on my RGH box and test these things for you though. It'll probably be tomorrow or the next day before I have results.
If you saw my information post on XboxChaos, You may have noticed ddx files are different for different games/platforms. Ddx textures for Oblivion (and FNV I think) have a different header and used zlib compression. I did not test this, I read someone else's findings. PS3 Skyrim uses a different header from Xbox360 and uses zlib compression. Xbox360 Skyrim uses the new XMem compression that I mentioned is used in the bsa archives. I am no encryption/compression guru: I researched Xentax.com and since I can't get the data to decompress using xbdecompress.exe, then supposedly it's the new Xmem compression which was introduced in XNA v4. There is an XNA v4 redist with runtimes that includes xnanative.dll. This dll has the functions to compress/decompress this new Xmem codec.
